graph-service/enricher.py

```
@@ -0,0 +1,909 @@
+    entity_type_lower = entity_type.lower().strip()
+    return any(eligible.lower() == entity_type_lower for eligible in eligible_types)
+
+def is_northdata_eligible_entity_type(entity_type: str) -> bool:
+    """Check if entity type is eligible for Northdata lookup (subset of enrichment-eligible types)."""
+    northdata_types = ["company", "organization", "person"]
+    entity_type_lower = entity_type.lower().strip()
+    return any(eligible.lower() == entity_type_lower for eligible in northdata_types)
+
+async def search_northdata(entity_name: str, browser: Browser) -> List[Dict[str, Any]]:
+    """Search Northdata for company/organization/person and extract result URLs."""
+    if not is_northdata_eligible_entity_type(entity_type):
```

Ruff (0.14.7)

Undefined name `entity_type`

(F821)